paasta_tools/cli/cmds/mark_for_deployment.py

```diff
@@ -107,9 +133,103 @@ def paasta_mark_for_deployment(args):
     if service and service.startswith('services-'):
         service = service.split('services-', 1)[1]
     validate_service_name(service, soa_dir=args.soa_dir)
-    return mark_for_deployment(
+    ret = mark_for_deployment(
         git_url=args.git_url,
         deploy_group=deploy_group,
         service=service,
         commit=args.commit,
     )
+    if args.block:
+        try:
+            _log(
+                service=service,
+                line="Waiting for deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+            wait_for_deployment(args.service, args.deploy_group,
+                                args.commit, args.soa_dir, args.timeout)
+            _log(
+                service=service,
+                line="Deployment of {0} to {1} complete".format(args.commit, deploy_group),
+                component='deploy',
+                level='event'
+            )
+        except TimeoutError:
```

Comment:
Nice 👍  I've actually mentioned something in when I catch the exception one level down but there's some good advice in their that I'll definitely add.

One thing I'm unsure about is the logging. At the moment I'm using `_log` which goes to paasta logs if I understand correctly. However, I guess it won't be shown in the Jenkins output. Ideally I'd like to see something in both.